client/src/main/java/com/vesoft/nebula/client/meta/MetaClient.java

```diff
@@ -114,8 +127,22 @@ public void close() {
      * @return
      */
     public synchronized List<IdName> getSpaces() throws TException {
+        int retry = RETRY_TIMES;
         ListSpacesReq request = new ListSpacesReq();
-        ListSpacesResp response = client.listSpaces(request);
+        ListSpacesResp response = null;
+        try {
+            while (retry-- >= 0) {
+                response = client.listSpaces(request);
+                if (response.getCode() == ErrorCode.E_LEADER_CHANGED) {
+                    freshClient(response.getLeader());
+                } else {
+                    break;
+                }
+            }
+        } catch (TException e) {
+            LOGGER.error(String.format("List Spaces Error: %s", e.getMessage()));
+            throw e;
+        }
         return response.getSpaces();
```

Comment:
need to check error is succeeded otherwise the `getSpaces()` is null.

Reply:
when error is not SUCCEED, if don't return null, then what do you want to return